### PR TITLE
Use `evalScript` to fill in benchmarks for failing programs

### PR DESCRIPTION
**Pull request by srid**

### CODE REVIEW

**Comment by srid** [comment]
CI passed, except for benchmark-diff failing. I should get rid of this diff thingy now...

**Comment by L-as** [comment]
This is technically correct, but you're not making full use of this. With this, we always have access to the used budget and can thus always include it.

**Comment by srid** [comment]
> This is technically correct, but you're not making full use of this. With this, we always have access to the used budget and can thus always include it.

I see what you mean ... okay done.

File: plutarch-benchmark/src/Plutarch/Benchmark.hs [hunk 1]
```diff
@@ -65,42 +66,46 @@ import qualified Plutus.V1.Ledger.Api as Plutus
...
+mkBenchmark :: ExBudget -> ScriptSizeBytes -> Benchmark
+mkBenchmark (ExBudget cpu mem) = Benchmark cpu mem
+
 newtype ScriptSizeBytes = ScriptSizeBytes Int64
   deriving stock (Eq, Ord, Show, Generic)
   deriving newtype (Num, ToField)
   deriving newtype (ToJSON)
 
+scriptSize :: Script -> ScriptSizeBytes
+scriptSize = ScriptSizeBytes . fromInteger . toInteger . SBS.length . serialiseScriptShort
```
Comment (L-as) [inline plutarch-benchmark/src/Plutarch/Benchmark.hs:104]: `fromInteger . toInteger`? `fromIntegral`?

File: plutarch-test/src/Plutarch/Test/Golden.hs [hunk 2]
```diff
@@ -168,17 +170,21 @@ currentGoldenKey = do
         Just path ->
           pure $ sconcat $ fmap GoldenKey path
 
-{- | Like `evaluateScript` but doesn't fail. Also returns `Script`.
+{- | Like `evalScript` but doesn't fail, and returns `Benchmark`.
```
Comment (L-as) [inline plutarch-test/src/Plutarch/Test/Golden.hs:173]: Comment seems outdated.

File: plutarch-benchmark/src/Plutarch/Benchmark.hs [hunk 3]
```diff
@@ -65,42 +66,46 @@ import qualified Plutus.V1.Ledger.Api as Plutus
...
+
+serialiseScriptShort :: Script -> SBS.ShortByteString
+serialiseScriptShort = SBS.toShort . LB.toStrict . serialise -- Using `flat` here breaks `evalScriptCounting`
```
Comment (L-as) [inline plutarch-benchmark/src/Plutarch/Benchmark.hs:107]: Can you elaborate on the comment BTW?

**Review by L-as — CHANGES_REQUESTED** [review]
Looks mostly good to me

File: plutarch-test/src/Plutarch/Test/Golden.hs (hunk 2 above)
Comment (srid) [reply plutarch-test/src/Plutarch/Test/Golden.hs:173]: It is accurate?

File: plutarch-benchmark/src/Plutarch/Benchmark.hs (hunk 3 above)
Comment (srid) [reply plutarch-benchmark/src/Plutarch/Benchmark.hs]: This looks to be an outdated comment; I remember trying to use the `flat` library for some reason, but it is not used anymore. Removed.

File: plutarch-test/src/Plutarch/Test/Golden.hs (hunk 2 above)
Comment (L-as) [reply plutarch-test/src/Plutarch/Test/Golden.hs:173]: `evalScript` doesn't fail either

Comment (srid) [reply plutarch-test/src/Plutarch/Test/Golden.hs]: By "faIl" I meant it returning `Left EvalError`; I've clarified the comment.

**Review by L-as — APPROVED** [review]
LGTM

**Pull request merged**